### DISCUSSION TITLE
local snapshotting, remote pulling?

### DISCUSSION THREAD

**Discussion by alaricljs** [psy0rz/zfs_autobackup#172] [Q&A]
I saw the split snapshot/push explanation [here](https://github.com/psy0rz/zfs_autobackup/wiki#splitting-up-snapshot-and-backup-job), but what I want to do is have my send/recv initiated remotely as a pull operation.  I'm just going to give it a try and see where I get but wanted to ask in case someone's done this before or it's a known bad idea.

Thanks

**Comment by psy0rz** [comment]
Yes, you can make offline snapshots on the source side and do pulling from the target side.

Just use it in snapshot mode on the source side, so it doesnt need to connect the target via ssh. (ill update the documentation to make this more clear)

**Comment by psy0rz** [comment]
I add this to https://github.com/psy0rz/zfs_autobackup/wiki#splitting-up-snapshot-and-backup-job:
```
You can also create the snapshots in offline mode by using zfs-autobackup as a snapshot tool on the source side. This way the snapshots will always be created, even if the backup server is offline or unreachable.
```

Does that clear it up for you? :)

**Reply by alaricljs** [reply]
Finally got a chance to try it out and it works as advertised.  IMHO it would be nice to have a flag to not manage (thin) snapshots on the source end when doing the backup transfer as a separate operation but I suppose `--keep-source 99999` is probably going to do it.

**Reply by psy0rz** [reply]
you can also use --no-thinning, so it wont destroy any snapshots.

**Reply by psy0rz** [reply]
however, sometimes its usefull to have the offline snapshotter do thinning as well, to prevent a built up of useless snapshots what will be destroyed anyway when the backup is started.
